lib/minidoc/finders.rb

```diff
@@ -24,17 +24,16 @@ def exists?(selector = {})
 
     def find(id_or_selector, options = {})
       if id_or_selector.is_a?(Hash)
-        options.merge!(transformer: method(:wrap))
-        collection.find(id_or_selector, options)
+        collection.find(id_or_selector, options).lazy.map(&method(:wrap))
```

Comment:
I think that's to be expected with this implementation, and that if you're interested in doing an atomic database count, you should use `Minidoc.count` rather than `Minidoc.find({...}).count`, which will execute the queries.

Ways to avoid that, that I can think of:

1.  re-introduce the transformer concept into the driver (or take a closer look and see if it's still there, but renamed to something else)
1. return a new type from this method, something like `Minidoc::FindResult`, which wraps the cursor, and generally behaves like a lazy enumerator, but delegates `count` to the wrapped cursor